lugodev/qvapay-python
# Add support for async functions and other things

# Changes

- Add support for `async`
- Tests with `100%` of code coverage
- Remove `dataclasses-json` dependency
- Add convection files for communities like `CODE_OF_CONDUCT.md` and others
- Add new banner and badges to `README.md`
- Update the content of `README.md` with new methods
- Remove `setup.py` and `requirements.txt` files because they are not necessary with poetry
- GitHub Action for CI (check pythonic style and run tests)
- GitHub Action for auto-publish in PyPi and GitHub Release when tag with v*.*.* be pushed
- Add migration guide section to `README.md`

# Breaking changes

- `from qvapay.v1 import *` instead of `from qvapay import *`
- `QvaPayClient` instead of `Client`
- `client.get_info` instead of `client.info`
- `client.get_balance` instead of `client.balance`
- `client.get_transactions` instead of `client.transactions`

# Tests:

For the tests to pass correctly, it is necessary that they can access the `app_id` and the `app_secret`:

- Remotely in the settings of GitHub repository two [Action Secrets](https://github.com/lugodev/qvapay-python/settings/secrets/actions) must be set: `QVAPAY_APP_ID` and `QVAPAY_APP_SECRET`
- Locally you must have a file placed in the root of the project called `.env` with the following structure

```
QVAPAY_APP_ID=...
QVAPAY_APP_SECRET=...
```

# Publish

To publish correctly in **PyPi**, two [Action Secrets](https://github.com/lugodev/qvapay-python/settings/secrets/actions) must be set in the settings of GitHub repository: `PYPI_USERNAME` and `PYPI_PASSWORD`

# Coverage

For the Coverage report in the README and in the PR it is necessary to log in to [codecov.io](https://codecov.io) and activate the repository, in addition to installing the Codecov GitHub App

# Note

In the migration guide in the README, when the new version is published, the version must be specified, right now it is set to `0.x.x`
